compiler/qsc_frontend/src/resolve.rs

```diff
@@ -108,6 +113,18 @@ pub(super) enum Error {
     #[diagnostic(help("this item is not implemented and cannot be used"))]
     #[diagnostic(code("Qsc.Resolve.Unimplemented"))]
     Unimplemented(String, #[label] Span),
+
+    #[error("this export is not a callable or type")]
+    #[diagnostic(code("Qsc.Resolve.ExportedNonItem"))]
+    ExportedNonItem(#[label] Span),
+
+    #[error("exporting external items is not yet supported")]
+    #[diagnostic(code("Qsc.Resolve.ExportedExternalItem"))]
+    ExportedExternalItem(#[label] Span),
+
+    #[error("export statements are only allowed in a namespace scope")]
+    #[diagnostic(code("Qsc.Resolve.ExportFromNonNamespaceScope"))]
+    ExportFromNonNamespaceScope(Span),
```

Comment:
We may want to tweak the error message here, for the user's sake. The word "namespace" won't make sense to them as we don't want them using the `namespace` keyword anymore, and they won't know what "implicit namespaces" are.

Maybe just state what we _don't_ want to happen, like:

```suggestion
    #[error("export statements are not allowed in a local scope")]
    #[diagnostic(code("Qsc.Resolve.ExportFromLocalScope"))]
    ExportFromLocalScope(Span),
```

or

```suggestion
    #[error("export statements are not allowed in a callable body")]
    #[diagnostic(code("Qsc.Resolve.ExportFromCallableBody"))]
    ExportFromCallableBody(Span),
```

or maybe something better if you can think of it.